crates/wasmparser/src/validator/names.rs

```diff
@@ -316,14 +509,144 @@ impl KebabName {
+                                slash: None,
+                                at: Some(At {
+                                    index: at as u32,
+                                    semver: Semver::Semver(ver),
+                                }),
+                            },
+                            Err(e) => {
+                                bail!(offset, "failed to parse version: {e}")
+                            }
+                        }
+                    } else {
+                        ParsedKebabName::RegistryId {
+                            colon: colon as u32,
+                            slash: None,
+                            at: None,
+                        }
+                    }
+                }
+            }
+            ComponentImportName::Unlocked(name) => {
+                let colon = find(name, ':')?;
+                validate_kebab(&name[..colon])?;
+                let slash = maybe_find(name, '/');
+                if let Some(sl) = slash {
+                    let at = name[sl..].find('@').map(|i| i + sl);
+                    validate_kebab(&name[colon + 1..sl])?;
+                    if let Some(at) = at {
+                        validate_kebab(&name[sl + 1..at])?;
+                        let version_string = validate_version_range(&name[at + 1..])?;
+                        ParsedKebabName::RegistryId {
+                            colon: colon as u32,
+                            slash: Some(sl as u32),
+                            at: Some(At {
+                                index: at as u32,
+                                semver: Semver::SemverRange(version_string),
+                            }),
+                        }
+                    } else {
+                        ParsedKebabName::RegistryId {
+                            colon: colon as u32,
+                            slash: Some(sl as u32),
+                            at: None,
+                        }
+                    }
+                } else {
+                    let at = name[colon..].find('@').map(|i| i + colon);
+                    if let Some(at) = at {
+                        validate_kebab(&name[colon + 1..at])?;
+                        let version_string = validate_version_range(&name[at + 1..])?;
+                        ParsedKebabName::RegistryId {
+                            colon: colon as u32,
+                            slash: None,
+                            at: Some(At {
+                                index: at as u32,
+                                semver: Semver::SemverRange(version_string),
+                            }),
+                        }
+                    } else {
+                        ParsedKebabName::RegistryId {
+                            colon: colon as u32,
+                            slash: None,
+                            at: None,
+                        }
+                    }
                 }
```

Comment:
I believe this is the more-or-less the same parsing code as `Locked`, so could those be deduplicated?